src/Service.Tests/SqlTests/GraphQLMutationTests/MultipleCreateMutationTests/MultipleCreateMutationTestBase.cs

```diff
@@ -0,0 +1,819 @@
+                                                    birthdate
+                                                }
+                                                }
+                                            }
+                                            }
+                                        }
+                                        ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true);
+            SqlTestHelper.PerformTestEqualJsonStrings(expectedResponse, actual.ToString());
+
+            // Validate that the records are created in the linking table
+            string actualResponseFromLinkingTable = await GetDatabaseResultAsync(linkingTableDbValidationQuery);
+            SqlTestHelper.PerformTestEqualJsonStrings(expectedResponseFromLinkingTable, actualResponseFromLinkingTable);
+        }
+
+        #endregion
+
+        #region Relationships defined through config file
+
+        /// <summary>
+        /// <code>Do: </code> Point create mutation with entities related through a 1:1 relationship. Relationship is defined through the config file.
+        /// <code>Check: </code> createUser_NonAutogenRelationshipColumn and UserProfile_NonAutogenRelationshipColumn items are successfully created in the database. UserProfile_NonAutogenRelationshipColumn item is created and linked in the database.
+        /// </summary>
+        public async Task MultipleCreateMutationWithOneToOneRelationshipDefinedInConfigFile(string expectedResponse1, string expectedResponse2)
+        {
+            // Point create mutation request with the related entity acting as referencing entity.
+            string graphQLMutationName = "createUser_NonAutogenRelationshipColumn";
+            string graphQLMutation1 = @"mutation {
+                  createUser_NonAutogenRelationshipColumn(
+                    item: {
+                      username: ""DAB""
+                      email: ""dab@microsoft.com""
+                      UserProfile_NonAutogenRelationshipColumn: {
+                        profilepictureurl: ""dab/profilepicture""
+                        userid: 10
+                      }
+                    }
+                  ) {
+                    userid
+                    username
+                    email
+                    UserProfile_NonAutogenRelationshipColumn {
+                      profileid
+                      userid
+                      username
+                      profilepictureurl
+                    }
+                  }
+                }";
+
+            JsonElement actualResponse1 = await ExecuteGraphQLRequestAsync(graphQLMutation1, graphQLMutationName, isAuthenticated: true);
+            SqlTestHelper.PerformTestEqualJsonStrings(expectedResponse1, actualResponse1.ToString());
+
+            // Point create mutation request with the top level entity acting as referencing entity.
```

Comment:
> with the top level entity acting as referencing entity.
what are the implications of this? 
- which entity is inserted first (UserProfile_...) and then second(User_...)?
- insert of UserProfile is source of truth for userId 10 that is provided to insert of User?

Reply:
Here, User.username and UserProfile.username are the relationship fields. Added comments to call out, relationship fields and insertion order.